src/lavinmq/mqtt/broker.cr

```diff
@@ -75,8 +75,7 @@ module LavinMQ
       end
 
       def subscribe(client, topics)
-        session = sessions[client.client_id]? || sessions.declare(client)
-        session.client = client
+        session = sessions[client.client_id]? || sessions.declare(client).tap { |s| s.client = client }
```

Comment:
Why doesn't the `declare` method do the assignment? it has the `client` available to it.

Reply:
oh yeah, thats true. lets do that